### PR TITLE
Michaelrmartin

### PR DESCRIPTION
**Pull request by michaelrmartin**
Adds hikes model and controller

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,10 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  # Hikes routes
+
+  get "/hikes" => "hikes#index"
+  get "/hikes/:id" => "hikes#show"
```
Comment (gatorjuice) [inline config/routes.rb:10]: you can use Rails' `resources` syntax here:

`resources :hikes, only: %i[index show]`

will give you the same ReSTful routes you have here.

File: db/migrate/20230605171154_create_hikes.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def change
+    create_table :hikes do |t|
+      t.string :name
+      t.string :length
+      t.text :description
+      t.string :city
+      t.string :region
+      t.string :country
+      t.string :lat
+      t.string :lon
```
Comment (gatorjuice) [inline db/migrate/20230605171154_create_hikes.rb:11]: If you plan to use the Ruby Geocoder library, these will be come conventional stored as `:latitude` and `:longitude` and their date type needs to either be `float`, or `decimal` but I personal prefer `decimal`

File: db/migrate/20230605171154_create_hikes.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      t.string :difficulty
```
Comment (gatorjuice) [inline db/migrate/20230605171154_create_hikes.rb:12]: Things such as difficulty or other categories are best served using Rails enum api. The datatype would be decimal but I can show you how it makes sense after you dress up the model properly.

File: app/controllers/hikes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  def show
+    hike_id = params["id"]
+    hike = Hike.find_by(id: hike_id)
+
+    render json: hike.as_json 
+
+  end
+
```
Comment (gatorjuice) [inline app/controllers/hikes_controller.rb:16]: it's a good practice to use "strong parameters" in your controllers where you "allow-list" parameter values 

```ruby
  def show_params
    params.require(:id)
  end
```

then in your controller method you use it...

```ruby
  def show
    hike = Hike.find(show_params[:id])

    render json: hike.as_json 
  end
```

If you use strong params and enforce what the controller expects to receive, you can get really good error messaging built in when a call isn't working, like maybe you're not building the request right in the front end.

**Review by gatorjuice — COMMENTED** [review]
App's off to a great start. Just some things I noticed.

File: config/routes.rb (hunk 1 above)
Comment (michaelrmartin) [reply config/routes.rb:10]: routes have been updated to reflect this

File: app/controllers/hikes_controller.rb (hunk 4 above)
Comment (michaelrmartin) [reply app/controllers/hikes_controller.rb:16]: updated with strong parameters.

File: db/migrate/20230605171154_create_hikes.rb (hunk 3 above)
Comment (michaelrmartin) [reply db/migrate/20230605171154_create_hikes.rb:12]: changed the model, but need to look more in to implementing Rails enum to update seeds data

**Review by avia62490 — APPROVED** [review]
Looks good, merging

**Pull request merged**